Champions/Taric/E.cs

```diff
@@ -0,0 +1,72 @@
+using GameServerCore.Enums;
+using GameServerCore.Domain.GameObjects;
+using static LeagueSandbox.GameServer.API.ApiFunctionManager;
+using LeagueSandbox.GameServer.GameObjects.AttackableUnits.AI;
+using GameServerCore.Domain;
+using LeagueSandbox.GameServer.Scripting.CSharp;
+
+namespace Spells
+{
+    public class Dazzle : IGameScript
+    {        
+        public void OnActivate(IChampion owner)
+        {
+        }
+
+        public void OnDeactivate(IChampion owner)
+        {
+        }
+
+        public void OnStartCasting(IChampion owner, ISpell spell, IAttackableUnit target)
+        {
+            
+        }
+
+        public void OnFinishCasting(IChampion owner, ISpell spell, IAttackableUnit target)
+        {            
+            if (owner.GetDistanceTo(target) > 625)
+            {
+                return;
+            }
+            else
+            {
+                spell.AddProjectileTarget("Dazzle", target, true);
+            }            
+        }
+
+        public void ApplyEffects(IChampion owner, IAttackableUnit target, ISpell spell, IProjectile projectile)
+        {
+            var time = 1.1f + 0.1f * spell.Level;
+            var ap = owner.Stats.AbilityPower.Total;
+            var damage = 10 + spell.Level * 30 + ap * 0.2f;
+            if (owner.GetDistanceTo(target) <= 460)
+            {
+                damage = 15 + spell.Level * 45 + ap * 0.3f;
+                if (owner.GetDistanceTo(target) <= 295)
+                {
+                    damage = 20 + spell.Level * 60 + ap * 0.4f;
+                }
+            }
+
+            target.TakeDamage(owner, damage, DamageType.DAMAGE_TYPE_MAGICAL, DamageSource.DAMAGE_SOURCE_SPELL, false);
+
+            AddBuffHudVisual("Stun", time, 1, BuffType.COMBAT_DEHANCER, (ObjAiBase)target, time);

```

Comment:
Seems that these 2 lines aren't yet ported to the API. This syntax is problematic as its using down casting and some lib functions.

Reply:
Well,So what can I fix it, I only know code like that, or I need to wait for API comes out (althought currently it works fine in game)

Reply:
TakeDamage and AddBuffHudVisual are in the API, but AddBuffHudVisual should be moved to a buff script.

Reply:
Right, So only HudVisual need to moved to Buff Script, I'll fix it

Reply:
Changed, and I forgot use a branch to PR it